.github/workflows/cypress-nightly.yml

```diff
@@ -10,63 +10,37 @@ on:
-          wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
-          sudo apt install --allow-downgrades -y ./google-chrome-stable_current_amd64.deb
+      - name: Build the docker containers
+        # Use a random string to avoid other providers on the same subnet which might cause tests to fail because it expects only providers named provider-1 and provider-2
+        run: docker compose -f tests/e2e/docker-compose.yml build
 
-      - name: Build the SDK
-        run: |
-          npm install
-          npm run build
+      - name: Start the docker containers
+        # Use a random string to avoid other providers on the same subnet which might cause tests to fail because it expects only providers named provider-1 and provider-2
+        run: docker compose -f tests/e2e/docker-compose.yml down && docker compose -f tests/e2e/docker-compose.yml up -d
 
-      - name: Configure python
-        continue-on-error: true
-        uses: actions/setup-python@v4
-        with:
-          python-version: "3.10"
+      - name: Fund the requestor
+        # Use a funding script which will retry funding the requestor 3 times, else it exits with error. The faucet is not reliable and sometimes fails to fund the requestor, thus the retry.
+        run: sleep 4 && docker exec -t e2e-requestor-1 /bin/sh -c "/golem-js/tests/e2e/fund-requestor.sh"
 
-      - name: Install goth
+      - name: Install and build the SDK in the docker container
         run: |
-          pip install goth
-          rm -rf ../goth/assets
-          python -m goth create-assets ../goth/assets
-          sed -Ezi 's/(use\-proxy:\s)(True)/\1False/mg' ../goth/assets/goth-config.yml
-          sed -Ezi 's/(use\-prerelease:\s)(false)/\1true\n    release-tag: "0.13.0-rc10"/mg' ../goth/assets/goth-config.yml
-          sed -i '/^ENTRYPOINT/i ENV YAGNA_AUTOCONF_APPKEY=try_golem' ../goth/assets/docker/yagna-goth-deb.Dockerfile
-
-      - name: Cleanup Docker
-        if: always()
-        run: |
-          c=$(docker ps -q) && [[ $c ]] && docker kill $c
-          docker system prune -af
-
-      - name: Log in to GitHub Docker repository
-        run: echo ${{ secrets.GITHUB_TOKEN }} | docker login docker.pkg.github.com -u ${{github.actor}} --password-stdin
+          docker exec -t e2e-requestor-1 /bin/sh -c "cd /golem-js && npm i && npm run build && ./node_modules/.bin/cypress install"
 
       - name: Run web server
         run: |
-          cd examples/web
-          node app.mjs &
+          docker exec -t -d e2e-requestor-1 /bin/sh -c "cd /golem-js/examples/web && node app.mjs"
 
       - name: Run test suite
-        env:
-          GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
-          npm run test:cypress -- --browser chrome
+          docker exec -t e2e-requestor-1 /bin/sh -c "cd /golem-js &&  npm run test:cypress -- --browser chromium"
```

Comment:
Chrome wasn't the simplest to install inside the docker container, but chromium was available as a package, so I just opted for the simplest solution. 

If we must use `chrome` then we need to update the Dockerfile and install the chrome driver.

Reply:
I used something like this before in our ci and it worked:
```
         sudo apt-get install -y libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
          wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
          sudo apt install --allow-downgrades -y ./google-chrome-stable_current_amd64.deb
```

Reply:
but if it's not easy, leave chromium

Reply:
It's mostly because Chrome has no ARM binaries for linux as far as I can see, and i'm testing this on my Macbook M1, so its troublesome for me to install. Alternatively I need to emulate x86 but then development is slow for me